pandas/core/generic.py

```diff
@@ -6863,6 +6863,18 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
             index = np.arange(len(_maybe_transposed_self._get_axis(alt_ax)))
         else:
             index = _maybe_transposed_self._get_axis(alt_ax)
+            methods = {"index", "values", "nearest", "time"}
+            is_numeric_or_datetime = (
+                is_numeric_dtype(index) or
+                is_datetime64_dtype(index) or
+                is_timedelta64_dtype(index)
+            )
+            if method not in methods and not is_numeric_or_datetime:
+                raise ValueError(
+                    "Index column must be numeric or datetime type when "
+                    "using any interpolation method other than linear. "
+                    "Try setting a numeric or datetome index column before "
+                    "interpolating.")
```

Comment:
you can add that ``{method}`` was passed in the text.

Reply:
use a replacement, e.g. "{method}"....format(method=method)

Reply:
done